### DISCUSSION TITLE
Special Registers

### DISCUSSION THREAD

**Discussion by aansaarii** [ChampSim/ChampSim#396] [General]
Hello!
The simulator uses special registers like REG_STACK_POINTER = 6, REG_FLAGS = 25, and REG_INSTRUCTION_POINTER = 26. Would you clarify to which particular ISA these registers belong? It seems that these are originally coming from the PIN-tool and hence x86 architecture. However, if I am not mistaken, IPC-1 traces were all from the ARM ISA. However, the simulator continues using these registers to identify the branch type. Would you clarify how these register assignments are used to identify branch types for an ARM ISA?
Thanks

**Comment by ngober** [comment]
I discussed this more in #382, but the constants `champsim::REG_STACK_POINTER` are descriptive of how traces have marked the stack pointer, etc. In order to be properly interpreted by the simulator, the trace must adhere to these. Whether this should continue to be the case (or if the trace should contain this information) is something under active discussion.
